pkg/util/azureclient/storage.go

```diff
@@ -34,3 +36,25 @@ func NewAccountsClient(ctx context.Context, subscriptionID string, authorizer au
 		AccountsClient: client,
 	}
 }
+
+// GetOrCreate will create the account if it doesn't exist, or return the account if already there
+func (c *accountsClient) GetOrCreate(ctx context.Context, resourceGroup, accountName string, createParams storage.AccountCreateParameters) (*storage.Account, error) {
```

Comment:
pkg/util/azureclient aims to transparently wrap existing Azure APIs without adding logic.  Currently the only exception is that we allow calling future.WaitForCompletionRef in our wrapper layer - see storage_addons.go as an example.  The level of logic add proposed here is not permissible.